control-plane-operator/controllers/hostedcontrolplane/manifests/secretproviderclass.go

```diff
@@ -0,0 +1,45 @@
+package manifests
+
+import (
+	"github.com/openshift/hypershift/support/azureutil"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	secretsstorev1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+)
+
+// SecretProviderClassForAroHCP returns an instance of a SecretProviderClass completed with its name, Azure Key Vault set
+// up, and the certificate name it needs to pull from the Key Vault.
+//
+// https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-identity-access?tabs=azure-portal&pivots=access-with-a-user-assigned-managed-identity
+func SecretProviderClassForAroHCP(secretProviderClassName, namespace, keyVaultName, KeyVaultTenantID, certificateName string) *secretsstorev1.SecretProviderClass {
```

Comment:
can we not add specific ref for ForAROHCP? This is a building block we happen to instantiate when installing in that env but it ins't necessarily exclusive to it. Should this be ManagedAzureKeyVaultSecretProviderClass or similar?